appveyor.yml

```diff
@@ -50,11 +50,18 @@ environment:
       BUILD_NAME: mingw
       EXTRA_BUILD_FLAGS: -j2
 
+    - CMAKE_GENERATOR: Unix Makefiles
+      BUILD_NAME: cygwin
+      EXTRA_BUILD_FLAGS: -j2
+
 install:
   - if not exist "%MINGW_ARCHIVE%" appveyor DownloadFile "%MINGW_URL%" -FileName "%MINGW_ARCHIVE%"
   - 7z x -y "%MINGW_ARCHIVE%" > nul
   - ps: Get-Command sh.exe -All | Remove-Item
   # CMake refuses to generate MinGW Makefiles if sh.exe is in the PATH
+  - if "%BUILD_NAME%"=="cygwin" if not exist "C:\cygwin64" mkdir "C:\cygwin64"
+  - if "%BUILD_NAME%"=="cygwin" appveyor DownloadFile "http://cygwin.com/setup-x86_64.exe" -FileName "C:\cygwin64\setup-x86_64.exe"
+  - if "%BUILD_NAME%"=="cygwin" C:\cygwin64\setup-x86_64.exe --quiet-mode --no-shortcuts --upgrade-also --root "C:\cygwin64" --local-package-dir "C:\cache\cygwin" --packages cmake,automake,make,gcc-core,gcc-g++
```

Comment:
AppVeyor build worker images are supposed to already come with Cygwin 2.10 (https://www.appveyor.com/docs/build-environment/#mingw-msys-cygwin). Are these lines necessary to install Cygwin-flavored build dependencies? (cmake, autotools, gcc, etc)